### DISCUSSION TITLE
port proxy via domain conflicts with code server

### DISCUSSION THREAD

**Discussion by geiseri** [coder/code-server#2899] [Q&A]
if i use the subdomain proxy approach with my external name i get an error.  my server is at `code.foo.local and domain to foo.local.  i get the error below:
![image](https://user-images.githubusercontent.com/2265693/84551738-7392bb00-acdc-11ea-8f1c-3d1374f33f02.png)
must the domain be a child as 1234.code.foo.local?

**Comment by nhooyr** [comment]
Should just work. cc @code-asher

**Comment by code-asher** [comment]
Oof, something unfortunate seems to have happened to the CSS, we'll have to look into that.

Yeah if you have `--proxy-domain foo.local` then it will try to proxy `*.foo.local` which includes `code.foo.local`.

That means you'll need to use `--proxy-domain code.foo.local` and browse to `<port>.code.foo.local` like you've noted, or use some other domain for the proxying if that feels too long.

The proxy feature uses the host header, so you could also set the host to `foo.local` when the domain is `code.foo.local`, then code-server won't proxy because it'll think it's on `foo.local`.

We should add a check to make sure the subdomain is a number before proxying to better support a same-level scenario like this.

**Comment by code-asher** [comment]
Oh right the CSS won't load because all the CSS requests will be having the same "not a valid number" error. So nevermind on that.
